types/matches.d.ts

```diff
@@ -1,6 +1,15 @@
+import {ARIARole} from 'aria-query'
+
 export type MatcherFunction = (content: string, element: HTMLElement) => boolean
 export type Matcher = string | RegExp | MatcherFunction
 
+// Allow `String` to have intellisense for roles while writing ByRole queries
+// If `string` is used it overrules the ARIARole string union types
+// resulting in no intellisense
+// We still want to support roles not included in the union type
+/* tslint:disable:ban-types */
+export type ByRoleMatcher = ARIARole | String | RegExp | MatcherFunction
```

Comment:
This use case is tracked in https://github.com/microsoft/TypeScript/issues/29729.

I'd prefer we go with the approach that is documented in that issue: `'literal' | (string & {})` (https://github.com/microsoft/TypeScript/issues/29729#issuecomment-505826972). Then we at least follow community best practices. And we should probably link the issue in the comment in case it gets resolved.

Reply:
🤯 I was looking for a workaround but I couldn't find any.
Thanks for also giving this a look @eps1lon , this is simply great!